compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -141,6 +141,10 @@ class PlainPrinter(_ctx: Context) extends Printer {
         ParamRefNameString(tp) ~ ".type"
       case tp: TypeParamRef =>
         ParamRefNameString(tp) ~ lambdaHash(tp.binder)
+      case tp: ConstantType =>
+        toTextRef(tp)
+      case tp: TermRef =>
+        toTextPrefix(tp.prefix) ~ selectionString(tp) ~ ".type"
```

Comment:
Why special case `TermRef`?

Reply:
This is for `a.type` singleton types.

```scala
scala> final val a = "foo"
val a: "foo" = foo

scala> val x: a.type = a
val x: a.type = foo
```
a.type is made by special casing TermRef. I am sorry if that's nonsensical 😛

Reply:
That makes some sense, but there are other cases above for `tp: TermRef` which already produce `toTextRef(tp) ~ ".type"` (which is equivalent to the code here), do they not apply?

Reply:
I had missed that. No, those cases (in PlainPrinter) are not applied because of the guards.